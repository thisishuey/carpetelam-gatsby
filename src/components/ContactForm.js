import React from "react";
import PropTypes from "prop-types";
import createStyles from "@material-ui/core/styles/createStyles";
import withStyles from "@material-ui/core/styles/withStyles";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";

function styles() {
  return createStyles({
    honeypot: {
      display: "none"
    }
  });
}

function GatsbyContactForm({ classes, named = {} }) {
  const { honeypot } = classes;
  const { name } = named;
  return (
    <form
      data-netlify="true"
      data-netlify-honeypot="phone"
      method="GET"
      name="contact-page-form"
      {...named}
    >
      <input type="hidden" name="form-name" value={name} />
      <Grid container spacing={16}>
        <Grid item sm={6} xs={12}>
          <TextField
            fullWidth
            id="contact-name"
            label="Name"
            name="name"
            placeholder="Princess Zelda"
            required
            variant="outlined"
          />
        </Grid>
        <Grid item sm={6} xs={12}>
          <TextField
            autoComplete="email"
            fullWidth
            id="contact-email"
            label="Email"
            name="email"
            placeholder="princess.zelda@hyrule.gov"
            type="email"
            required
            variant="outlined"
          />
        </Grid>
        <Grid className={honeypot} item xs={12}>
          <TextField
            fullWidth
            id="contact-phone"
            label="Phone"
            name="phone"
            placholder="404.555.1212"
            variant="outlined"
          />
        </Grid>
        <Grid item xs={12}>
          <TextField
            fullWidth
            id="contact-message"
            label="Message"
            multiline
            name="message"
            placeholder="Open your eyes... Wake up, Link."
            required
            rows="4"
            variant="outlined"
          />
        </Grid>
        <Grid item sm={3} xs={12}>
          <Button
            color="primary"
            fullWidth
            margin="normal"
            size="large"
            type="submit"
            variant="contained"
          >
            Send Email
          </Button>
        </Grid>
      </Grid>
    </form>
  );
}

GatsbyContactForm.propTypes = {
  children: PropTypes.string,
  classes: PropTypes.object,
  named: PropTypes.object
};

export default withStyles(styles)(GatsbyContactForm);
