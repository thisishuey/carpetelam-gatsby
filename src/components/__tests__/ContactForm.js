import React from "react";
import { render } from "react-testing-library";
import ContactForm from "../ContactForm";

const { container, getByLabelText } = render(<ContactForm />);

test("contact form matches snapshot", () => {
  expect(container.firstChild).toMatchSnapshot();
});

test("contact form renders name field", () => {
  expect(getByLabelText(/name/i)).toBeDefined();
});

test("contact form renders email field", () => {
  expect(getByLabelText(/email/i)).toBeDefined();
});

test("contact form renders message field", () => {
  expect(getByLabelText(/message/i)).toBeDefined();
});
