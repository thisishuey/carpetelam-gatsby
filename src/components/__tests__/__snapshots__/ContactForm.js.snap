// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact form matches snapshot 1`] = `
<form
  data-netlify="true"
  data-netlify-honeypot="phone"
  method="GET"
  name="contact-page-form"
>
  <input
    name="form-name"
    type="hidden"
    value=""
  />
  <div
    class="MuiGrid-container-2 MuiGrid-spacing-xs-16-25"
  >
    <div
      class="MuiGrid-item-3 MuiGrid-grid-xs-12-42 MuiGrid-grid-sm-6-50"
    >
      <div
        class="MuiFormControl-root-99 MuiFormControl-fullWidth-102"
      >
        <label
          class="MuiFormLabel-root-114 MuiFormLabel-required-119 MuiInputLabel-required-107 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-outlined-113"
          data-shrink="false"
        >
          Name
          <span
            class="MuiFormLabel-asterisk-120"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
        >
          <fieldset
            aria-hidden="true"
            class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
            style="padding-left: 8px;"
          >
            <legend
              class="MuiPrivateNotchedOutline-legend-152"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
          <input
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiOutlinedInput-input-129"
            name="name"
            placeholder="Princess Zelda"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-3 MuiGrid-grid-xs-12-42 MuiGrid-grid-sm-6-50"
    >
      <div
        class="MuiFormControl-root-99 MuiFormControl-fullWidth-102"
      >
        <label
          class="MuiFormLabel-root-114 MuiFormLabel-required-119 MuiInputLabel-required-107 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-outlined-113"
          data-shrink="false"
        >
          Email
          <span
            class="MuiFormLabel-asterisk-120"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
        >
          <fieldset
            aria-hidden="true"
            class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
            style="padding-left: 8px;"
          >
            <legend
              class="MuiPrivateNotchedOutline-legend-152"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
          <input
            aria-invalid="false"
            autocomplete="email"
            class="MuiInputBase-input-144 MuiOutlinedInput-input-129 MuiInputBase-inputType-147"
            name="email"
            placeholder="princess.zelda@hyrule.gov"
            required=""
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-3 MuiGrid-grid-xs-12-42 GatsbyContactForm-honeypot-1"
    >
      <div
        class="MuiFormControl-root-99 MuiFormControl-fullWidth-102"
        placholder="404.555.1212"
      >
        <label
          class="MuiFormLabel-root-114 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-outlined-113"
          data-shrink="false"
        >
          Phone
        </label>
        <div
          class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
        >
          <fieldset
            aria-hidden="true"
            class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
            style="padding-left: 8px;"
          >
            <legend
              class="MuiPrivateNotchedOutline-legend-152"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
          <input
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiOutlinedInput-input-129"
            name="phone"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-3 MuiGrid-grid-xs-12-42"
    >
      <div
        class="MuiFormControl-root-99 MuiFormControl-fullWidth-102"
      >
        <label
          class="MuiFormLabel-root-114 MuiFormLabel-required-119 MuiInputLabel-required-107 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-outlined-113"
          data-shrink="false"
        >
          Message
          <span
            class="MuiFormLabel-asterisk-120"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135 MuiInputBase-multiline-142 MuiOutlinedInput-multiline-127"
        >
          <fieldset
            aria-hidden="true"
            class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
            style="padding-left: 8px;"
          >
            <legend
              class="MuiPrivateNotchedOutline-legend-152"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiOutlinedInput-input-129 MuiInputBase-inputMultiline-146 MuiOutlinedInput-inputMultiline-131"
            name="message"
            placeholder="Open your eyes... Wake up, Link."
            required=""
            rows="4"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-item-3 MuiGrid-grid-xs-12-42 MuiGrid-grid-sm-3-47"
    >
      <button
        class="MuiButtonBase-root-179 MuiButton-root-153 MuiButton-contained-164 MuiButton-containedPrimary-165 MuiButton-raised-167 MuiButton-raisedPrimary-168 MuiButton-sizeLarge-177 MuiButton-fullWidth-178"
        margin="normal"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-154"
        >
          Send Email
        </span>
        <span
          class="MuiTouchRipple-root-182"
        />
      </button>
    </div>
  </div>
</form>
`;
